### ISSUE TITLE
Loyalty Program Issues

### ISSUE THREAD

**Issue by Zlash65** [frappe/erpnext#14847]
- [ ] Loyalty point deduction should be done on `net_total` and not on `grand_total`

- [ ] In Customer master there is a read only field `Loyalty Program Tier`. What is its purpose ?

- [ ] If auto-opt-in is enabled then should it be applied to Customer when Sales Invoice is being made ? Right now, if a new Customer is made and there if there is any Loyalty Program with auto-opt-in enabled, then that Program is automatically selected while Customer's creation. So for already existing customer, there should be some provision right ?

- [ ] There should be a threshold to how much one can redeem ?

- [ ] Minimum spent set in Loyalty Program does not seem to to have been enforced

- [ ] If in Loyalty Program, 'Customer Group' & 'Territory' are not set then does that mean it should be applicable to Customer for all kinds ? Right now there's a strict check with 'and' conditioning. So if I were to have a customer whose 'Customer Group' or 'Territory' or both is not set, then no Loyalty Program would be applied for it ( for auto-opt-in). Manually I'll be able to add any Loyalty Program to it.

**Comment by olamide2** [comment]
I agree with everything @Zlash65  stated above with the exception of ¨There should be a threshold to how much one can redeem ?¨.

We do not need to have a minimum or maximum amount to redeem.

However all the other issues raised are valid concerns that need to be addressed.

Regards

**Comment by kossanah** [comment]
@olamide2  there some instances the business would not want to allow a 100% discount by the customer using coupon point earned.

**Comment by nabinhait** [comment]
@all Are you sure loyalty points should always be earned based on Net Total (Total without taxes)?

**Comment by gupteshwar** [comment]
There are certain more cases (based on Loyalty Program we run for a customer) which I think are important and for your reference,
- In my case products are MRP based (inclusive of taxes) and loyalty is calculated only on actual payment amount done through a Cash Mode/Credit Card/Any mode by which actual currency is spent. This excludes payment by Credit Note/Coupons/Vouchers/Loyalty Redemption. In a way works on Grand Total but actually only on bank/cash currency payment.
- We should have setting whether this to work on Net or Grand and as well whether on specific payment mode or all.
- Every Loyalty Points Earned by a Customer should have an expiry days/date associated. (This is crucial as ultimately it's a parallel currency and a liability on company).
- Either through redemption or expiry we should maintain right liability at any point in time.
- On Sales Returns either company may offer Credit Note in that case points should not revert back but otherwise this should reverse back.
- There has to be detailed log around every point (since it follows a FIFO) how much been consumed? (more importantly when partial so as to expire those appropriately). How much been auto-expired?
